app/services/current-session.js

```diff
@@ -46,4 +47,8 @@ export default class CurrentSessionService extends Service {
       throw new Error('Group classification not loaded');
     return this.groupClassification.label;
   }
+
+  setCurrentAssociation(association) {
```

Comment:
I wouldn't overload the current session service with data not related with the current session, as the current association selected, it would be cleaner to make another service for this if you want to go with that approach.
But anyway the current association selected should be available via the model in all the routes that are a child of the association route